File: CS/HowToUseCriteriaPropertyEditors.Module/CriteriaController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;

using DevExpress.ExpressApp.Editors;


namespace HowToUseCriteriaPropertyEditors.Module {
    public partial class CriteriaController : ViewController {
        public CriteriaController() {
            InitializeComponent();
            RegisterActions(components);
        }

        private void CriteriaController_Activated(object sender, EventArgs e) {
            FilteringCriterionAction.Items.Clear();
            FilteringCriterionAction.Items.Add(new ChoiceActionItem("All the products", null));
            foreach (FilteringCriterion criterion in (View as ListView).ObjectSpace.GetObjects<FilteringCriterion>(null)) {
                FilteringCriterionAction.Items.Add(new ChoiceActionItem(criterion.Description, criterion.Criterion));
            }
            FilteringCriterionAction.SelectedIndex = 0;
        }

        private void FilteringCriterionAction_Execute(object sender, SingleChoiceActionExecuteEventArgs e) {
            (View as ListView).CollectionSource.Criteria.Clear();
            (View as ListView).CollectionSource.Criteria[e.SelectedChoiceActionItem.Caption] =
               CriteriaEditorHelper.GetCriteriaOperator(e.SelectedChoiceActionItem.Data as string, typeof(Product), View.ObjectSpace);
        }
    }
}

File: VB/HowToUseCriteriaPropertyEditors.Module/CriteriaController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.Data.Filtering

Namespace HowToUseCriteriaPropertyEditors.Module
    Partial Public Class CriteriaController
        Inherits ViewController
        Public Sub New()
            InitializeComponent()
            RegisterActions(components)
        End Sub
        Private Sub CriteriaController_Activated(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Activated
            FilteringCriterionAction.Items.Clear()
            FilteringCriterionAction.Items.Add(New ChoiceActionItem("All the products", Nothing))
            For Each criterion As FilteringCriterion In (TryCast(View, ListView)).ObjectSpace.GetObjects(Of FilteringCriterion)(Nothing)
                FilteringCriterionAction.Items.Add(New ChoiceActionItem(criterion.Description, criterion.Criterion))
            Next criterion
            FilteringCriterionAction.SelectedIndex = 0
        End Sub
        Private Sub FilteringCriterionAction_Execute(ByVal sender As Object, ByVal e As SingleChoiceActionExecuteEventArgs) Handles FilteringCriterionAction.Execute
            TryCast(View, ListView).CollectionSource.Criteria.Clear()
            TryCast(View, ListView).CollectionSource.Criteria(e.SelectedChoiceActionItem.Caption) = CriteriaEditorHelper.GetCriteriaOperator(TryCast(e.SelectedChoiceActionItem.Data, String), GetType(Product), View.ObjectSpace)
        End Sub
    End Class
End Namespace
